pkg/build/util/generate.go

```diff
@@ -154,3 +160,19 @@ func SubstituteImageReferences(build *buildapi.Build, oldImage string, newImage
 		}
 	}
 }
+
+// GetBuildPodName returns name of the build pod.
+func GetBuildPodName(build *buildapi.Build) string {
+	return fmt.Sprintf("pod-%s", build.Name)
+}
+
+// GetNextBuildName returns name of the next build and increments BuildConfig's LastVersion.
+func GetNextBuildName(bc *buildapi.BuildConfig) string {
+	bc.LastVersion++
+	return fmt.Sprintf("%s-%d", bc.Name, bc.LastVersion)
+}
+
+// getNextBuildNameFromBuild returns name of the next build with random uuid added at the end
+func getNextBuildNameFromBuild(build *buildapi.Build) string {
+	return fmt.Sprintf("%s-%s", build.Name, uuid.NewUUID().String())
```

Comment:
This applies only to Build generated from Build, initially I wanted that to have some additional incremental mechanism but I failed with not enough time to do it nicely. I'll change that UUID to some timestamp and make sure we fit into 63 char limit. Thx @csrwng.